### PR TITLE
Added mac variant of portuguese keymap

### PR DESCRIPTION
**Pull request by edevil**
This keymap is very similar to the existing portuguese keymap, but
some symbols are moved around.

## Description

New keymap, for the mac variant of the Portuguese keymap.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [X] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [X] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [X] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [X] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.
- [X] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by drashna** [comment]
Just a heads up: 

If the changes involve just control, alt, and/or gui, it may be simpler to use bootmagic/magic instead of this.

File: docs/ja/reference_keymap_extras.md [hunk 1]
```diff
@@ -51,6 +51,7 @@ HID Keyboard/Keypad usage ページでは 256 未満の usage が定義されて
 | Norwegian                   | `keymap_norwegian.h`             |
 | Polish                      | `keymap_polish.h`                |
 | Portuguese                  | `keymap_portuguese.h`            |
+| Portuguese (mac)            |`keymap_portuguese_mac.h`         |
```
Comment (fauxpark) [inline docs/ja/reference_keymap_extras.md:54]:
```suggestion
| Portuguese (macOS, ISO)     | `keymap_portuguese_mac_iso.h`    |
```

File: docs/reference_keymap_extras.md [hunk 2]
```diff
@@ -46,6 +46,7 @@ To use these, simply `#include` the corresponding [header file](https://github.c
 |Norwegian                  |`keymap_norwegian.h`            |
 |Polish                     |`keymap_polish.h`               |
 |Portuguese                 |`keymap_portuguese.h`           |
+|Portuguese (mac)           |`keymap_portuguese_mac.h`       |
```
Comment (fauxpark) [inline docs/reference_keymap_extras.md:49]:
```suggestion
|Portuguese (macOS, ISO)    | `keymap_portuguese_mac_iso.h`  |
```

File: quantum/keymap_extras/keymap_portuguese_mac.h [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+#define PTM_Z    KC_Z    // Z
+#define PTM_X    KC_X    // X
+#define PTM_C    KC_C    // C
+#define PTM_V    KC_V    // V
+#define PTM_B    KC_B    // B
+#define PTM_N    KC_N    // N
+#define PTM_M    KC_M    // M
+#define PTM_COMM KC_COMM // ,
+#define PTM_DOT  KC_DOT  // .
+#define PTM_MINS KC_SLSH // -
```
Comment (fauxpark) [inline quantum/keymap_extras/keymap_portuguese_mac.h:87]:
```suggestion
#define PT_SECT KC_NUBS // §
#define PT_1    KC_1    // 1
#define PT_2    KC_2    // 2
#define PT_3    KC_3    // 3
#define PT_4    KC_4    // 4
#define PT_5    KC_5    // 5
#define PT_6    KC_6    // 6
#define PT_7    KC_7    // 7
#define PT_8    KC_8    // 8
#define PT_9    KC_9    // 9
#define PT_0    KC_0    // 0
#define PT_QUOT KC_MINS // '
#define PT_PLUS KC_EQL  // +
// Row 2
#define PT_Q    KC_Q    // Q
#define PT_W    KC_W    // W
#define PT_E    KC_E    // E
#define PT_R    KC_R    // R
#define PT_T    KC_T    // T
#define PT_Y    KC_Y    // Y
#define PT_U    KC_U    // U
#define PT_I    KC_I    // I
#define PT_O    KC_O    // O
#define PT_P    KC_P    // P
#define PT_MORD KC_LBRC // º
#define PT_ACUT KC_RBRC // ´ (dead)
// Row 3
#define PT_A    KC_A    // A
#define PT_S    KC_S    // S
#define PT_D    KC_D    // D
#define PT_F    KC_F    // F
#define PT_G    KC_G    // G
#define PT_H    KC_H    // H
#define PT_J    KC_J    // J
#define PT_K    KC_K    // K
#define PT_L    KC_L    // L
#define PT_CCED KC_SCLN // Ç
#define PT_TILD KC_QUOT // ~ (dead)
#define PT_BSLS KC_NUHS  // (backslash)
// Row 4
#define PT_LABK KC_GRV  // <
#define PT_Z    KC_Z    // Z
#define PT_X    KC_X    // X
#define PT_C    KC_C    // C
#define PT_V    KC_V    // V
#define PT_B    KC_B    // B
#define PT_N    KC_N    // N
#define PT_M    KC_M    // M
#define PT_COMM KC_COMM // ,
#define PT_DOT  KC_DOT  // .
#define PT_MINS KC_SLSH // -
```

File: quantum/keymap_extras/keymap_portuguese_mac.h [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+#define PTM_FORD S(PTM_MORD) // ª
+#define PTM_GRV  S(PTM_ACUT) // ` (dead)
+// Row 3
+#define PTM_CIRC S(PTM_TILD) // ^ (dead)
+#define PTM_PIPE S(PTM_BSLS) // |
+// Row 4
+#define PTM_RABK S(PTM_LABK) // >
+#define PTM_SCLN S(PTM_COMM) // ;
+#define PTM_COLN S(PTM_DOT)  // :
+#define PTM_UNDS S(PTM_MINS) // _
```
Comment (fauxpark) [inline quantum/keymap_extras/keymap_portuguese_mac.h:126]:
```suggestion
#define PT_PLMN S(PT_BSLS) // ±
#define PT_EXLM S(PT_1)    // !
#define PT_DQUO S(PT_2)    // "
#define PT_HASH S(PT_3)    // #
#define PT_DLR  S(PT_4)    // $
#define PT_PERC S(PT_5)    // %
#define PT_AMPR S(PT_6)    // &
#define PT_SLSH S(PT_7)    // /
#define PT_LPRN S(PT_8)    // (
#define PT_RPRN S(PT_9)    // )
#define PT_EQL  S(PT_0)    // =
#define PT_QUES S(PT_QUOT) // ?
#define PT_ASTR S(PT_PLUS) // *
// Row 2
#define PT_FORD S(PT_MORD) // ª
#define PT_GRV  S(PT_ACUT) // ` (dead)
// Row 3
#define PT_CIRC S(PT_TILD) // ^ (dead)
#define PT_PIPE S(PT_BSLS) // |
// Row 4
#define PT_RABK S(PT_LABK) // >
#define PT_SCLN S(PT_COMM) // ;
#define PT_COLN S(PT_DOT)  // :
#define PT_UNDS S(PT_MINS) // _
```

File: quantum/keymap_extras/keymap_portuguese_mac.h [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+ * ├────┼───┴┬──┴─┬─┴───┴───┴───┴───┴───┴──┬┴───┼───┴┬────┬────┤
+ * │    │    │    │                        │    │    │    │    │
+ * └────┴────┴────┴────────────────────────┴────┴────┴────┴────┘
+ */
+// Row 1
+#define PTM_AT   ALGR(PTM_2)    // @
+#define PTM_EURO ALGR(PTM_3)    // €
+#define PTM_PND  ALGR(PTM_4)    // £
+#define PTM_LBRC ALGR(PTM_8)    // [
+#define PTM_RBRC ALGR(PTM_9)    // ]
```
Comment (fauxpark) [inline quantum/keymap_extras/keymap_portuguese_mac.h:146]:
```suggestion
#define PT_AT   ALGR(PT_2)    // @
#define PT_EURO ALGR(PT_3)    // €
#define PT_PND  ALGR(PT_4)    // £
#define PT_LBRC ALGR(PT_8)    // [
#define PT_RBRC ALGR(PT_9)    // ]
```

File: quantum/keymap_extras/keymap_portuguese_mac.h [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+ * │      │   │   │   │   │   │   │   │   │   │   │   │   │    │
+ * ├────┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴───┴────┤
+ * │    │   │   │   │   │   │   │   │   │   │   │   │          │
+ * ├────┼───┴┬──┴─┬─┴───┴───┴───┴───┴───┴──┬┴───┼───┴┬────┬────┤
+ * │    │    │    │                        │    │    │    │    │
+ * └────┴────┴────┴────────────────────────┴────┴────┴────┴────┘
+ */
+// Row 1
+#define PTM_LCBR S(ALGR(PTM_8)) // {
+#define PTM_RCBR S(ALGR(PTM_9)) // }
```
Comment (fauxpark) [inline quantum/keymap_extras/keymap_portuguese_mac.h:163]:
```suggestion
#define PT_LCBR S(ALGR(PT_8)) // {
#define PT_RCBR S(ALGR(PT_9)) // }
```

**Review by fauxpark — CHANGES_REQUESTED** [review]
This should define the entire set of Alted and Shift+Alted symbols, see `keymap_italian_osx_iso.h`.

File: docs/ja/reference_keymap_extras.md (hunk 1 above)
Comment (edevil) [reply docs/ja/reference_keymap_extras.md]: I implemented this change.

**Comment by edevil** [comment]
Renamed file to `keymap_portuguese_osx_iso.h` and added the rest of the symbols.

File: quantum/keymap_extras/keymap_portuguese_osx_iso.h [hunk 7]
```diff
@@ -0,0 +1,235 @@
...
+#define PT_OMEG ALGR(PT_Z)    // Ω
+#define PT_LDAQ ALGR(PT_X)    // «
+#define PT_COPY ALGR(PT_C)    // ©
+#define PT_SQRT ALGR(PT_V)    // √
+#define PT_INTG ALGR(PT_B)    // ∫
+#define PT_NOT  ALGR(PT_N)    // ¬
+#define PT_MICR ALGR(PT_M)    // µ
+#define PT_LDQU ALGR(PT_COMM) // “
+#define PT_ELLP ALGR(PT_DOT)  // …
+#define PT_MDSH ALGR(PT_MINS) // —
```
Comment (fauxpark) [inline quantum/keymap_extras/keymap_portuguese_osx_iso.h:190]:
```suggestion
#define PT_LTEQ A(PT_LABK) // ≤
#define PT_OMEG A(PT_Z)    // Ω
#define PT_LDAQ A(PT_X)    // «
#define PT_COPY A(PT_C)    // ©
#define PT_SQRT A(PT_V)    // √
#define PT_INTG A(PT_B)    // ∫
#define PT_NOT  A(PT_N)    // ¬
#define PT_MICR A(PT_M)    // µ
#define PT_LDQU A(PT_COMM) // “
#define PT_ELLP A(PT_DOT)  // …
#define PT_MDSH A(PT_MINS) // —
```

File: quantum/keymap_extras/keymap_portuguese_osx_iso.h [hunk 8]
```diff
@@ -0,0 +1,235 @@
...
+#define PT_FL   S(ALGR(PT_3))    // ﬂ
+#define PT_CENT S(ALGR(PT_4))    // ¢
+#define PT_INFN S(ALGR(PT_5))    // ∞
+#define PT_BULT S(ALGR(PT_6))    // •
+#define PT_FRSL S(ALGR(PT_7))    // ⁄
+#define PT_LCBR S(ALGR(PT_8))    // {
+#define PT_RCBR S(ALGR(PT_9))    // }
+#define PT_AEQL S(ALGR(PT_0))    // ≈
+#define PT_IQUE S(ALGR(PT_QUOT)) // ¿
+#define PT_LOZN S(ALGR(PT_PLUS)) // ◊
```
Comment (fauxpark) [inline quantum/keymap_extras/keymap_portuguese_osx_iso.h:217]:
```suggestion
#define PT_IEXL S(A(PT_1))    // ¡
#define PT_FI   S(A(PT_2))    // ﬁ
#define PT_FL   S(A(PT_3))    // ﬂ
#define PT_CENT S(A(PT_4))    // ¢
#define PT_INFN S(A(PT_5))    // ∞
#define PT_BULT S(A(PT_6))    // •
#define PT_FRSL S(A(PT_7))    // ⁄
#define PT_LCBR S(A(PT_8))    // {
#define PT_RCBR S(A(PT_9))    // }
#define PT_AEQL S(A(PT_0))    // ≈
#define PT_IQUE S(A(PT_QUOT)) // ¿
#define PT_LOZN S(A(PT_PLUS)) // ◊
```

File: quantum/keymap_extras/keymap_portuguese_osx_iso.h [hunk 9]
```diff
@@ -0,0 +1,235 @@
...
+#define PT_RSQU S(ALGR(PT_L))    // ’
+#define PT_OGON S(ALGR(PT_CCED)) // ˛
+#define PT_DCIR S(ALGR(PT_TILD)) // ˆ (dead)
+#define PT_RSAQ S(ALGR(PT_BSLS)) // ›
+// Row 4
+#define PT_GTEQ S(ALGR(PT_LABK)) // ≥
+#define PT_RDAQ S(ALGR(PT_X))    // »
+#define PT_RDQU S(ALGR(PT_COMM)) // ”
+#define PT_MDDT S(ALGR(PT_DOT))  // ·
+#define PT_NDSH S(ALGR(PT_MINS)) // –
```
Comment (fauxpark) [inline quantum/keymap_extras/keymap_portuguese_osx_iso.h:235]:
```suggestion
#define PT_DDAG S(A(PT_U))    // ‡
#define PT_RNGA S(A(PT_I))    // ˚
#define PT_NARP S(A(PT_P))    // ∏
#define PT_DACU S(A(PT_ACUT)) // ˝
// Row 3
#define PT_INCR S(A(PT_D))    // ∆
#define PT_SLQU S(A(PT_K))    // ‚
#define PT_RSQU S(A(PT_L))    // ’
#define PT_OGON S(A(PT_CCED)) // ˛
#define PT_DCIR S(A(PT_TILD)) // ˆ (dead)
#define PT_RSAQ S(A(PT_BSLS)) // ›
// Row 4
#define PT_GTEQ S(A(PT_LABK)) // ≥
#define PT_RDAQ S(A(PT_X))    // »
#define PT_RDQU S(A(PT_COMM)) // ”
#define PT_MDDT S(A(PT_DOT))  // ·
#define PT_NDSH S(A(PT_MINS)) // –
```

File: quantum/keymap_extras/keymap_portuguese_osx_iso.h [hunk 10]
```diff
@@ -0,0 +1,235 @@
...
+#define PT_COPY ALGR(PT_C)    // ©
+#define PT_SQRT ALGR(PT_V)    // √
+#define PT_INTG ALGR(PT_B)    // ∫
+#define PT_NOT  ALGR(PT_N)    // ¬
+#define PT_MICR ALGR(PT_M)    // µ
+#define PT_LDQU ALGR(PT_COMM) // “
+#define PT_ELLP ALGR(PT_DOT)  // …
+#define PT_MDSH ALGR(PT_MINS) // —
+
+/* Shift+AltGr symbols
```
Comment (fauxpark) [inline quantum/keymap_extras/keymap_portuguese_osx_iso.h:192]:
```suggestion
/* Shift+Alted symbols
```

File: quantum/keymap_extras/keymap_portuguese_osx_iso.h [hunk 11]
```diff
@@ -0,0 +1,235 @@
...
+// Row 3
+#define PT_CIRC S(PT_TILD) // ^ (dead)
+#define PT_PIPE S(PT_BSLS) // |
+// Row 4
+#define PT_RABK S(PT_LABK) // >
+#define PT_SCLN S(PT_COMM) // ;
+#define PT_COLN S(PT_DOT)  // :
+#define PT_UNDS S(PT_MINS) // _
+
+/* AltGr symbols
```
Comment (fauxpark) [inline quantum/keymap_extras/keymap_portuguese_osx_iso.h:128]:
```suggestion
/* Alted symbols
```

**Review by fauxpark — CHANGES_REQUESTED** [review]
Mostly stylistic changes here. Could you also make sure that all the "space" characters inside the box drawings are actually NBSP characters (Option+Space)? That way they still look nice even if you have whitespace turned on:
image

**Comment by edevil** [comment]
Thank you for the review @fauxpark . I think I found all the spaces and converted to nbsp.

File: quantum/keymap_extras/keymap_portuguese_osx_iso.h [hunk 12]
```diff
@@ -0,0 +1,234 @@
...
+#define PT_DOT  KC_DOT  // .
+#define PT_MINS KC_SLSH // -
+
+/* Shifted symbols
+ * ┌───┬───┬───┬───┬───┬───┬───┬───┬───┬───┬───┬───┬───┬─────┐
+ * │ ± │ ! │ " │ # │ $ │ % │ & │ / │ ( │ ) │ = │ ? │ * │     │
+ * ├───┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬───┤
+ * │     │   │   │   │   │   │   │   │   │   │   │ ª │ ` │   │
+ * ├─────┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┐  │
+ * │      │   │   │   │   │   │   │   │   │   │   │ ^ │ ^ │  │
```
Comment (fauxpark) [inline quantum/keymap_extras/keymap_portuguese_osx_iso.h:95]:
```suggestion
 * │      │   │   │   │   │   │   │   │   │   │   │ ^ │ | │  │
```

File: quantum/keymap_extras/keymap_portuguese_osx_iso.h [hunk 13]
```diff
@@ -0,0 +1,234 @@
...
+#define PT_PIPE S(PT_BSLS) // |
+// Row 4
+#define PT_RABK S(PT_LABK) // >
+#define PT_SCLN S(PT_COMM) // ;
+#define PT_COLN S(PT_DOT)  // :
+#define PT_UNDS S(PT_MINS) // _
+
+/* Alted symbols
+ * ┌───┬───┬───┬───┬───┬───┬───┬───┬───┬───┬───┬───┬───┬─────┐
+ * │   │  │ @ │ € │ £ │ ‰ │ ¶ │ ÷ │ [ │ ] │ ≠ │ § │   │     │
```
Comment (fauxpark) [inline quantum/keymap_extras/keymap_portuguese_osx_iso.h:130]:
```suggestion
 * │   │  │ @ │ € │ £ │ ‰ │ ¶ │ ÷ │ [ │ ] │ ≠ │   │   │     │
```

File: quantum/keymap_extras/keymap_portuguese_osx_iso.h [hunk 14]
```diff
@@ -0,0 +1,234 @@
...
+ * ├───┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬───┤
+ * │     │ œ │ ∑ │ æ │ ® │ ™ │ ¥ │ † │ ı │ ø │ π │ ° │ ¨ │   │
```
Comment (fauxpark) [inline quantum/keymap_extras/keymap_portuguese_osx_iso.h:132]:
```suggestion
 * │     │ Œ │ ∑ │ Æ │ ® │ ™ │ ¥ │ † │ ı │ Ø │ π │ ° │ ¨ │   │
```

File: quantum/keymap_extras/keymap_portuguese_osx_iso.h [hunk 15]
```diff
@@ -0,0 +1,234 @@
...
+ * ├─────┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┐  │
+ * │      │ å │ ß │ ∂ │ ƒ │ ˙ │ ˇ │ ¯ │ „ │ ‘ │ ¸ │ ˜ │ ‹ │  │
```
Comment (fauxpark) [inline quantum/keymap_extras/keymap_portuguese_osx_iso.h:134]:
```suggestion
 * │      │ Å │ ß │ ∂ │ ƒ │ ˙ │ ˇ │ ¯ │ „ │ ‘ │ ¸ │ ˜ │ ‹ │  │
```

**Review by fauxpark — CHANGES_REQUESTED** [review]
Some last little comment tweaks, then it looks good to go. If you'd like, you can also add the ANSI variant, and the sendstring LUT headers for both.

**Comment by edevil** [comment]
I committed your fixes. As for an ANSI version, I don't have an ANSI keyboard to try it on. Regarding the sendstring files, to be honest, I looked at a few of those files but I did not understand how they are used or how to build one myself. Is there some documentation about it?

**Comment by fauxpark** [comment]
The way I test other layouts is to simply go to the Keyboard preferences and hit the "Change keyboard type" button, from there if you spam a key the wizard will just give up and ask you to choose which layout you want to set it to. Then in the input sources tab you can see a preview of the layout which changes when you hold Shift and/or Option.

The sendstring LUT headers can be added to make `SEND_STRING()` macros work with non-US ANSI layouts set in your OS, and consist of three parts:

* `ascii_to_keycode_lut` - this maps the base (unmodified) keycodes to their equivalents in the ASCII table, for example in the French AZERTY layout the position for the `0` and  `@` characters in ASCII are set to `FR_AGRV` (which is actually `KC_0`) because `FR_0` and `FR_AT` are `S(FR_AGRV)` and `ALGR(FR_AGRV)`, respectively.
* `ascii_to_shift_lut` and `ascii_to_altgr_lut` - these are the same idea as above, except that they just mark keycodes that must be shifted or AltGr'ed to achieve the desired character in the ASCII table. For example all the positions for capital letters are set to 1 in the shift LUT. The `KCLUT_ENTRY()` macro is used to pack each row of 8 bits into a single byte to save space.

**Comment by edevil** [comment]
I added the sendstring file for the ISO version. I'll leave ANSI versions of these for a later time, I don't think they see much use anyway.

**Comment by fauxpark** [comment]
Your force push undid some of my suggestions. Other than that, looks good.

**Comment by edevil** [comment]
Ops, forgot to pull first. Now everything should be fine.

File: quantum/keymap_extras/keymap_portuguese_osx_iso.h [hunk 16]
```diff
@@ -127,11 +127,11 @@
 
 /* Alted symbols
  * ┌───┬───┬───┬───┬───┬───┬───┬───┬───┬───┬───┬───┬───┬─────┐
- * │   │  │ @ │ € │ £ │ ‰ │ ¶ │ ÷ │ [ │ ] │ ≠ │ § │   │     │
+ * │   │  │ @ │ € │ £ │ ‰ │ ¶ │ ÷ │ [ │ ] │ ≠ │   │   │     │
```
Comment (fauxpark) [inline quantum/keymap_extras/keymap_portuguese_osx_iso.h:130]: Sorry, this should be an NBSP. Either I didn't hit Option+Space or GitHub suggestions don't let you do it.

Comment (edevil) [reply quantum/keymap_extras/keymap_portuguese_osx_iso.h]: Done! :)

**Review by fauxpark — APPROVED** [review]

**Comment by edevil** [comment]
@drashna The test failures seem unrelated to this PR. Can this be merged?

**Review by drashna — APPROVED** [review]

**Comment by drashna** [comment]
Absolutely.  Thanks!

**Pull request merged**